libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -1577,12 +1577,19 @@ bool AP_Mission::mission_cmd_to_mavlink_int(const AP_Mission::Mission_Command& c
 
     // set defaults
     packet.current = 0;     // 1 if we are passing back the mission command that is currently being executed
+    packet.frame = 0;
+    packet.autocontinue = 1;
+
+    // Strictly these should be defaulted to NaN
     packet.param1 = 0;
     packet.param2 = 0;
     packet.param3 = 0;
     packet.param4 = 0;
-    packet.frame = 0;
-    packet.autocontinue = 1;
+
+    // Strictly these should be defaulted to INT32_MAX for x and y and NaN for z
+    packet.x = 0;
```

Comment:
Maybe just zero it - that way if extensions fields are added they also get zeroed...

Reply:
The caller may have filled in the target system and component ID. The changes in this function are not needed in anycase, the all the callers do pass in a zeroed structure (after the first commit). The main thing is at least zeroing these extras manually means that we always set the same fields no matter the way-point type.

Reply:
In fact the `get_item` fills in the `target_system`, `target_component` and `seq` before passing it in. It probably doesn't need to....  but it currently does....

Reply:
I have re-worked to zero and remove those unneeded sets.

Reply:
Hum, that broke everything because mission protocol uses `convert_MISSION_ITEM_INT_to_MISSION_ITEM`, I have reworked some more, but I'm tempted to go back to the just manually zeroing X Y and Z.